docker/Dockerfile

```diff
@@ -108,6 +108,11 @@ RUN mkdir /rust /cargo
 
 RUN curl -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain stable
 
+
+# arm64 builds consume a lot of memory if `CARGO_NET_GIT_FETCH_WITH_CLI` is not
+# set to true, so we expose it as a build-arg.
+ARG CARGO_NET_GIT_FETCH_WITH_CLI
```

Comment:
do ARGs become environment variables for the directives below them? (otherwise how does this work?!)

Reply:
Yup, I believe so!

Reply:
Good point, it doesn't look like it from a quick local test I've done.

Reply:
Good point, it doesn't look like it from a quick local test:

```
[18:09:36] babolivier@Noah:~/Documents/test$ cat Dockerfile 
FROM alpine
ARG TEST
COPY ./test.sh /test.sh
ENTRYPOINT ["/test.sh"]
[18:10:28] babolivier@Noah:~/Documents/test$ cat test.sh 
#!/bin/sh
env
[18:10:40] babolivier@Noah:~/Documents/test$ docker build -t test -f Dockerfile --build-arg=TEST=foo .
Sending build context to Docker daemon  3.072kB
Step 1/4 : FROM alpine
 ---> 9c6f07244728
Step 2/4 : ARG TEST
 ---> Using cache
 ---> 69700947c516
Step 3/4 : COPY ./test.sh /test.sh
 ---> Using cache
 ---> 2bb37955a594
Step 4/4 : ENTRYPOINT ["/test.sh"]
 ---> Using cache
 ---> 120f3318464d
Successfully built 120f3318464d
Successfully tagged test:latest
[18:12:11] babolivier@Noah:~/Documents/test$ docker run test --build-arg=TEST=foo
HOSTNAME=fce46e554943
SHLVL=1
HOME=/root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
[18:12:17] babolivier@Noah:~/Documents/test$ 
```

Reply:
nyargh what

Reply:
Oh for gods sake, `RUN $TEST` would do the right thing, but we actually need an `ENV TEST=$TEST` I think